src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -1048,6 +1057,199 @@ bool needsReporting()
+                    errorName: handlerType.Name,
+                    errorLocation,
+                    suppressResultDiagnostics: true,
+                    BindingDiagnosticBag.Discarded,
+                    memberResolutionResult: out _,
+                    candidateConstructors: out _,
+                    allowProtectedConstructorsOfBaseType: false);
+            }
+
+            if (success)
+            {
+                arguments.GetOrCreateData<ParameterWellKnownAttributeData>().InterpolatedStringHandlerArguments = parameterOrdinals;
+            }
+            else
+            {
+                // Could not find a constructor on '{0}' that matches the parameters specified in the InterpolatedStringHandlerArgumentAttribute.
+                diagnostics.Add(ErrorCode.ERR_CouldNotFindApplicableInterpolatedStringHandlerConstructor, errorLocation, handlerType);
+            }
+
+            (int Ordinal, ParameterSymbol? Parameter)? decodeName(TypedConstant constant, ref DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments)
+            {
+                Debug.Assert(arguments.AttributeSyntaxOpt is not null);
+                if (constant.IsNull)
+                {
+                    // null is not a valid parameter name. To get access to the receiver of an instance method, use the empty string as the parameter name.
+                    diagnostics.Add(ErrorCode.ERR_NullInvalidInterpolatedStringHandlerArgumentName, arguments.AttributeSyntaxOpt.Location);
+                    return null;
+                }
+
+                if (constant.TypeInternal is not { SpecialType: SpecialType.System_String })
+                {
+                    // There has already been an error reported. Just return null.
+                    return null;
+                }
+
+                var name = constant.DecodeValue<string>(SpecialType.System_String);
+                Debug.Assert(name != null);
+                if (name == "")
+                {
+                    // Name refers to the "this" instance parameter.
+                    if (ContainingSymbol.IsStatic)
+                    {
+                        // '{0}' is not an instance method, the receiver cannot be an interpolated string handler argument.
+                        diagnostics.Add(ErrorCode.ERR_NotInstanceInvalidInterpolatedStringHandlerArgumentName, arguments.AttributeSyntaxOpt.Location, ContainingSymbol);
+                        return null;
+                    }
+
+                    return (-1, null);
+                }
+
+                ImmutableArray<ParameterSymbol> parameters = ContainingSymbol switch
+                {
+                    PropertySymbol { Parameters: var p } => p,
+                    MethodSymbol { Parameters: var p } => p,
+                    _ => throw ExceptionUtilities.UnexpectedValue(ContainingType)
+                };
+
+                foreach (var parameter in parameters)
```

Comment:
>foreach (var parameter in parameters) [](http://example.com/codeflow?start=16&length=37)

It looks like for similar task for PEParameterSymbol we are using FirstOrDefault helper